src/main/scala/beam/agentsim/agents/Population.scala

```diff
@@ -44,20 +48,38 @@ class Population(
       case _: Exception      => Stop
       case _: AssertionError => Stop
     }
-  initHouseholds()
 
   override def receive: PartialFunction[Any, Unit] = {
     case TriggerWithId(InitializeTrigger(_), triggerId) =>
-      sender ! CompletionNotice(triggerId, Vector())
-    case Terminated(_) =>
-    // Do nothing
-    case Finish =>
-      context.children.foreach(_ ! Finish)
-      dieIfNoChildren()
-      context.become {
-        case Terminated(_) =>
-          dieIfNoChildren()
-      }
+      implicit val timeout: Timeout = Timeout(120, TimeUnit.SECONDS)
+      sharedVehicleFleets.foreach(_ ! GetVehicleTypes())
+      context.become(getVehicleTypes(triggerId, sharedVehicleFleets.size, Set.empty))
+  }
+
+  def getVehicleTypes(triggerId: Long, responsesLeft: Int, vehicleTypes: Set[BeamVehicleType]): Receive = {
+    if (responsesLeft <= 0) {
+      finishInitialization(triggerId, vehicleTypes)
+    } else {
+      case VehicleTypesResponse(sharedVehicleTypes) =>
+        context.become(getVehicleTypes(triggerId, responsesLeft - 1, vehicleTypes ++ sharedVehicleTypes))
```

Comment:
Yes, it would be stuck in this state. But in case of an exception in one of the shared vehicle fleet actors, we need to debug and fix this. In this case, the simulation would not function correctly anyway.
But I agree that it's better to add some error output in this case. How can I do it here?

Reply:
The errors would be from elsewhere, so that wouldn't help. I was referring to something like a timer so that if we haven't heard back after x time then it logs that only x of y have been heard back